engine/src/repository/game.rs

```diff
@@ -18,20 +20,30 @@ pub enum GameError {
     CreateGameFailed,
 }
 
-impl LaerningToolRepository {
-    pub async fn create_game(&self, game: Game) -> Result<Game, GameError> {
+#[async_trait]
+impl Repository<Game, GameError> for LaerningToolRepository {
+    async fn create(&self, game: Game) -> Result<Game, RepositoryError<GameError>> {
         let data: Option<Game> = self
             .db
             .query(r#"INSERT INTO game $game"#)
             .bind(("game", &game))
             .await
+            .map_err(|_| RepositoryError::CreationFailed(GameError::CreateGameFailed))
             .unwrap()
             .take(0)
             .unwrap();
-        data.ok_or(GameError::CreateGameFailed)
+
+        Ok(data.unwrap())
+    }
+
+    async fn create_batch(&self, datasets: Vec<Game>) -> Result<(), RepositoryError<GameError>> {
+        for dataset in datasets {
+            self.create(dataset).await.unwrap();
+        }
+        Ok(())
     }
 
-    pub async fn list_games(&self) -> Result<Vec<Game>, GameError> {
+    async fn list_nature(&self) -> Result<Vec<Game>, RepositoryError<GameError>> {
```

Comment:
I'm not sure what nature means over here. Could you please explain, as I haven't encountered this before?

Reply:
I'm categorizing Games, Datasets as Nature (thought of "Universe" as well). So instead of `list_game` or `list_dataset`, `list_nature` (or `list_universe`) should replace it.

what's your thoughts on it